engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/CardinalityAggregator.scala

```diff
@@ -84,5 +87,20 @@ private[aggregate] abstract class CardinalityWrapper extends Value with KryoSeri
     in.read(bytes)
     wrapped = readFromBytes(bytes)
   }
+
+  def canEqual(other: Any): Boolean = other.isInstanceOf[CardinalityWrapper]
+
+  override def equals(other: Any): Boolean = other match {
```

Comment:
can we use some builder for that, e.g. from commons? Is this null safe?

Reply:
This generated by Idea equals is null safe because null isn't handled by: `case that: CardinalityWrapper =>` but by default case. hashcode wasn't null-safe - I've replaced with `Objects.hash`.